lib/oelite/runq.py

```diff
@@ -557,6 +557,11 @@ def get_runabletask(self):
             return None
         return self.cookbook.get_task(id=task_id)
 
+    def get_runabletasks(self):
+        self.update_runabletasks()
+        ret = [self.cookbook.get_task(id=task_id) for task_id in self.runable]
+        self.runable = []
```

Comment:
Not my invention. Look at how get_runabletask used to be implemented; self.runable is a sort of cache which gets filled on every call (via a call to get_readytasks, which does some sql stuff), and then a single element is popped and returned. All I do is steal the entire cache.